platform-core/src/main/kotlin/org/dashj/platform/sdk/platform/MockPlatformStateRepository.kt

```diff
@@ -0,0 +1,274 @@
+                                found = true
+                                break
+                            }
+                        }
+                    }
+                    found
+                }
+
+                else -> throw UnsupportedOperationException("in has unknown operands: " + a.javaClass.name + " and " + b?.javaClass?.name)
+            }
+
+            ">" -> return when {
+                a is Int && b is Int -> b.toInt() > a.toInt()
+                a is Long && b is Long -> b.toLong() > a.toLong()
+                a is Int && b is Long -> b.toInt() > a.toLong()
+                a is Long && b is Int -> b.toLong() > a.toInt()
+                else -> throw throw UnsupportedOperationException("> has unknown operands: " + a.javaClass.name + " and " + b?.javaClass?.name)
+            }
+
+            else -> {
+                throw UnsupportedOperationException(operator)
+            }
+        }
+    }
+
+    private fun matchWhere(where: List<Any>, document: Document): Boolean {
+        val field = where[0] as String
+        val operator = where[1] as String
+        val value = where[2]
+        val rawDocument = document.toObject()
+        if (field.startsWith("\$")) {
+            val docValue = rawDocument[field]
+            return compare(value, docValue, operator)
+        } else {
+            val docValue = document.get(field)
+            return compare(value, docValue, operator)
+        }
+    }
+
+    override fun fetchDocuments(contractId: Identifier, documentType: String, where: Any): List<Document> {
+        val query = where as DocumentQuery
+        val byType = documentsMap.filter {
+            it.value.dataContractId == contractId && it.value.type == documentType
+        }
+
+        val results = byType.filter { entry ->
+            val document = entry.value
+            if (query.where == null) {
+                true
+            } else {
+                query.where!!.map { whereClause ->
+                    whereClause as List<*>
+                    matchWhere(whereClause as List<Any>, document)
+                }.all { queryResult ->
+                    queryResult
+                }
+            }
+        }
+        return results.values.toList()
+    }
+
+    override fun storeIdentity(identity: Identity) {
+        if (!identityMap.containsKey(identity.id)) {
+            identityMap[identity.id] = identity
+        }
+        storeIdentityPublicKeyHashes(identity.id, identity.publicKeys.map {
+            when (it.type) {
+                IdentityPublicKey.Type.ECDSA_HASH160, IdentityPublicKey.Type.BIP13_SCRIPT_HASH -> it.data
+                IdentityPublicKey.Type.ECDSA_SECP256K1 -> ECKey.fromPublicOnly(it.data).pubKeyHash
+                else -> Sha256Hash.twiceOf(it.data).bytes
+            }
+        })
+        log.info("store identity: {}", identity.toBuffer().toHex())
+    }
+
+
+    // for Mocking
+    private fun loadMockData() {
+
```

Comment:
This is where we add mock data.  Data is in CBOR format and can be taken from the logs of DashPay.

Currently there are two usernames that are created with profiles in this data.  And one ContactRequest.